### ISSUE TITLE
Markdown issue

### ISSUE THREAD

**Issue by JokerQyou** [gogits/gogs#2058]
The following content will not be rendered correctly.
```markdown
This is a sentence with [a link][1] and [another][2].

  [1]: http://a.com
  [2]: http://b.com
```

Instead, a space is needed for correctly rendering of the second link, like this:

```markdown
This is a sentence with [a link][1] and [another][2].

  [1]: http://a.com 
  [2]: http://b.com 
```

Notice there is a space after each of the two links.

**Comment by fanningert** [comment]
Ok, I also need a second look where the problem is. :)

After the `[1]: http://a.com` is as space (` `) in the second example. Also at the second link of example two.

**Comment by JokerQyou** [comment]
I can confirm that both of the links should have a tailing space, otherwise the second link could not be rendered correctly.

**Comment by Unknwon** [comment]
Thanks your feedback!

But feels like a bug of [blackfriday](https://github.com/russross/blackfriday).

**Comment by JokerQyou** [comment]
Hmm, should I report this directly to `blackfriday` or leave it for you to report to them?

**Comment by Unknwon** [comment]
@JokerQyou I think it's better let you report to blackfriday and reference to this issue so we can keep track of status.

**Comment by KesusKim** [comment]
I found some very difference with markdown out there and gogs' markdown. I am not a good markdowner, but do write something sometimes.

As I do like forking git repo to my gogs-hosted git repo, I quite did a lot, and found out, some README file is broken. (for example, in Github README, Header syntax can be working without empty space between # and title, but in Gogs README, it seems it must have empty space between # and title, like ##License only works in Github README, but not works in Gogs README. Checks out by migrating; https://github.com/junku901/machine_learning)

I don't know how to address these issues as I am very beginner, but I do feel little awkward when myself write down README file, too. Personally, I checks my markdown using stackeditor with very basic syntax, but when I move that markdown to my README file and push it, it is somehow different. So have to re-pushing over and over until I fix it.

I use Gogs version:  0.8.25, I don't know will that be a problem, but hope it would be fixed sooner or later.

**Comment by JokerQyou** [comment]
Some other git services act the same. I've seen @KesusKim 's problem a long time ago on Gitcafe (they have shutdown their website now).
Personally I prefer Github's way of parsing, which works if you happened to forget that space between hash tag and your text. But I did see all the examples on [Markdown: Syntax](https://daringfireball.net/projects/markdown/syntax) have a space in between.

**Comment by bkcsoft** [comment]
If I recall correctly, there's a settings in `app.ini => [markdown]` for
this (check `conf/app.ini` in the gogs-repo for the complete list of
settings)

**Comment by Unknwon** [comment]
@bkcsoft no... only hard line breaks are possible because blackfriday provides as an option, not for others.
